grpc_api.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * File: grpc_api.go
+ *
+ * Description: various funcs needed for the gRPC API
+ */
+
+package main
+
+import (
+	pb "./lclusterpb"
+	"golang.org/x/net/context"
+	"strconv"
+)
+
+//! API function to start a job
+/*
+ * @param     Context            given host context
+ * @param     StartJobRequest    object to hold the start job request
+ *
+ * @return    StartJobResponse   the server's response to the remote call
+ * @return    error              error message, if any
+ */
+func (s *LclusterdServer) StartJob(ctx context.Context,
+	r *pb.StartJobRequest) (*pb.StartJobResponse, error) {
+
+	// create a start job response husk for use later on...
+	response := &pb.StartJobResponse{}
+
+	// Cast the job request into a Job, then attempt to add it to the
+	// queue.
+	pid, err := etcdServer.addToGlobalQueue((*Job)(r))
+
+	// if any errors occur...
+	if err != nil || pid < 0 {
+
+		// set the pid to -1 and grab the error as a string
+		response.Pid = -1
+		response.Error = err.Error()
+
+		// then return the failed job creation via response
+		return response, err
+	}
+
+	// otherwise use the given pid and pass it back
+	response.Pid = pid
+	return response, nil
+}
+
+//! API function to check a job
+/*
+ * @param     Context            given host context
+ * @param     CheckJobRequest    object to hold the check job request
+ *
+ * @return    CheckJobResponse   the server's response to the remote call
+ * @return    error              error message, if any
+ */
+func (s *LclusterdServer) CheckJob(ctx context.Context,
+	cjr *pb.CheckJobRequest) (*pb.CheckJobResponse, error) {
+
+	// input validation
+	if cjr == nil || cjr.Pid < 1 {
```

Comment:
When I use protoc to generate code from the proto for the Lclusterd grpc functions, they seem to spawn code that utilizes pointers by default, so accepting pointers in those functions is the simplest way of handling it. That said, you bring up a good point, a few of the other functions use pointers needlessly, so I'll fix that.

https://github.com/rbisewski/lclusterd/commit/bec2e522b934791c8557f4b6217a32366e7e77d6